Alternation engine

AlexBlokh opened the pull request.

Defman commented:
How can I test the changes in this PR? I was wondering if a preview deployment is available through `https://pkg.pr.new` or if I should pull it down to test locally.

AndriiSherman commented:
> How can I test the changes in this PR? I was wondering if a preview deployment is available through `https://pkg.pr.new` or if I should pull it down to test locally.

It's not available yet because there are around 40-50 failing tests. Once those are fixed, this PR will be automatically deployed to a preview `npm` tag that can be used. We'll announce it right after all the tests pass

Defman commented:
> It's not available yet because there are around 40-50 failing tests. Once those are fixed, this PR will be automatically deployed to a preview `npm` tag that can be used. We'll announce it right after all the tests pass

Thank you.

I just had a quick look at the PR and noticed that CockroachDB Time Travel Quries are not supported, just wanted to note it here so it could perhaps be added to out stading issues are not planned, there might be other looking for the feature.

kravetsone commented:
V1 is great news! The Drizzle team is so amazing 
You guys work really hard...
how much will Drizzle maintain the development pace after V1? will the "stable major" interfere with any important changes? in any case, this is a good sign, and more teams will be able to try the worst library

Eigilak commented:
Goood damn you bois are doing gods work 🫡

AlexBlokh committed:
+

Sukairo-02 committed:
Ignored empty tests

Sukairo-02 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
+

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

Sukairo-02 committed:
todo -> skip

Sukairo-02 committed:
Added bunfig to avoid tests exit 1 bug

Sukairo-02 committed:
Temporarily disabled bun workflow tests

AlexBlokh committed:
+

AlexBlokh committed:
Merge commit '4f31ad4ac640ba2aa16643244b14cbef4d50f383' into alternation-engine

AlexBlokh committed:
Merge commit 'cb95e80f4991dd2eee86bfd5fbf24d1273494f64' into alternation-engine

RomanNabukhotnyi committed:
Enhance MySQL connection query to include type casting

AlexBlokh committed:
+

AlexBlokh committed:
ignore prisma in integrations tests

AlexBlokh committed:
+

AlexBlokh committed:
+

AlexBlokh committed:
+

OleksiiKH0240 committed:
[integration-tests] fixed seed tests

AlexBlokh committed:
add postgres postgis

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

AlexBlokh committed:
+

AlexBlokh committed:
+

AlexBlokh committed:
pull dbs simultaneously

AlexBlokh committed:
increase test timeout for exports/imports tests

AlexBlokh committed:
+

AlexBlokh committed:
chunck imports tests by 10

AlexBlokh committed:
massive perf boost for imports tests

AlexBlokh committed:
fix mysql tests

AlexBlokh committed:
fix pg tests

AlexBlokh committed:
+

AlexBlokh committed:
+

OleksiiKH0240 committed:
[integration-tests] updated gel tests

OleksiiKH0240 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
fix up postgres snapshot

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

AlexBlokh committed:
fix types

AlexBlokh committed:
remove console log

AleksandrSherman committed:
[cockroach-update]: updated not null handling

AlexBlokh committed:
add postgis for kit tests

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
cockroach speed up

AlexBlokh committed:
+

AlexBlokh committed:
+

AlexBlokh committed:
fix cockrch

AlexBlokh committed:
cockroach more concurrent tests

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

RomanNabukhotnyi committed:
Prepare merge 'api-studio' to 'alternation-engine'

AleksandrSherman committed:
[update]: added isExisting to schemas

RomanNabukhotnyi committed:
Refactor import statements to use type imports in various API files and update server start options for better configurability

AlexBlokh committed:
postgres kit:push schemas behaviour change

after a thorough discussion we decided to proceed with 'intuitive first':
- if I declare schemas in drizzle schema and I don't specify schemaFilter in config,
  all non-existing schemas (excluding 'public') will be pushed in db
- if I declare schemas in drizzle schema and I DO specify schemaFilter in config,
  only those schemas will be pushed, if any of them is missing in the schema files,
  an error will be thrown. If you'd want kit to ignore them - use .existing() method on schema instance.

AlexBlokh committed:
fix

AleksandrSherman committed:
[update]: schemas isExisting flag made public

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

OleksiiKH0240 committed:
[integration-tests] updated singlestore tests

OleksiiKH0240 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
upd feature branch attw pnpm installs

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

OleksiiKH0240 committed:
[drizzle-seed] removed docker container creation from pg-postgis data types tests

OleksiiKH0240 committed:
[int] added test

OleksiiKH0240 committed:
[drizzle-kit] updated mysql tests

Sukairo-02 committed:
Reworked `Stop DBs` to logic like `Start DBs`

Sukairo-02 committed:
Prevented `Stop DBs` when no DBs specified

Sukairo-02 committed:
Syntax fix

OleksiiKH0240 committed:
[drizzle-kit] added tests

Sukairo-02 committed:
Added case for `gel` to `Stop DBs`

Sukairo-02 committed:
`Stop DBs` rework

Sukairo-02 committed:
added `postgres-vector` container to `Stop DBs`

AlexBlokh committed:
remove stop dbs

AlexBlokh committed:
fix mysql diff

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

OleksiiKH0240 committed:
[drizzle-kit] updated tests

OleksiiKH0240 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AndriiSherman committed:
Added v3 folders migration

AndriiSherman committed:
Update husky

AndriiSherman committed:
Update husky

AndriiSherman committed:
check for pre-commit

AlexBlokh committed:
if pnpm is unavailable in husky hook - fallback to dockerized lint

AlexBlokh committed:
Merge commit '780a1da2523acca0741fa5d540621f8b7e3386a9' into alternation-engine

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

AndriiSherman committed:
check docker

AndriiSherman committed:
test 2

AlexBlokh committed:
fix lint.sh script to do lint-staged

AlexBlokh committed:
fix lint.sh

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
reimplement schemas in drizzle kit config

AlexBlokh committed:
up typescript target

Sukairo-02 committed:
Fixed incorrect usage of interval in test case

L-Mario564 committed:
Fix linting issues

L-Mario564 committed:
Remove ESLint files

AlexBlokh committed:
fix crdb

AlexBlokh committed:
Merge commit 'b96bff9b557d06a36bd0c26073f19d90fafbb2ef' into alternation-engine

L-Mario564 committed:
Fix lint errors

L-Mario564 committed:
Lint ignore line in test

AlexBlokh committed:
+

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

L-Mario564 committed:
Fix Kit tests

L-Mario564 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
+

AlexBlokh committed:
fix ci script

AlexBlokh committed:
Merge commit 'bbbe2226ac600315d66e77dc15e5dee2768c4aad' into alternation-engine

AlexBlokh committed:
typo fix

L-Mario564 committed:
Fix

L-Mario564 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

L-Mario564 committed:
Fix

AlexBlokh committed:
keep singlestore separated

AlexBlokh committed:
Merge commit 'cbe869af1d76fabc1a10779a7d6a8c6a1bf21c8c' into alternation-engine

AlexBlokh committed:
fix publish script

AlexBlokh committed:
fix version fetch

AlexBlokh committed:
remove redundant `npm version`

AlexBlokh committed:
fix

AlexBlokh committed:
:'(

AlexBlokh committed:
test no shared dist

AlexBlokh committed:
revert

AlexBlokh committed:
upload all artifacts in 1 job

AlexBlokh committed:
+

AlexBlokh committed:
+

AlexBlokh committed:
+

AlexBlokh committed:
we no longer need npm token for publishing

AlexBlokh committed:
try latest npm version

AlexBlokh committed:
ci fix

AlexBlokh committed:
I hate npm and github and CIs

AlexBlokh committed:
nuke .npmrc folder

AlexBlokh committed:
don't explicitely create tag for package

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

AndriiSherman committed:
V3folder (#5031)

* Added v3 folders migration

* Update husky

* Update husky

* check for pre-commit

* check docker

* test 2

* Fixes after manual folders checks

AlexBlokh committed:
debug

AlexBlokh committed:
ci+

AlexBlokh committed:
:'(

AlexBlokh committed:
+

Sukairo-02 committed:
Removed `.getSQL()` from tables and views, moved `enableRLS()` from pg table instances to `pgTable.withRLS(...)`

Sukairo-02 committed:
Updated kit tests to use new `withRLS()` syntax

Sukairo-02 committed:
Additional test case fixes, updated `cockroach` `withRLS()` syntax

AlexBlokh committed:
entity filters rework

made sure all flows respect both drizzle kit config entity filter and
.existing() entities in all dialects

AlexBlokh committed:
Merge commit '2129bd5ceda27d73e4d2263c07f42dd6cf36c024' into alternation-engine

AlexBlokh committed:
fix .withRLS ts introspect

AlexBlokh committed:
squash attw steps into 1

AlexBlokh committed:
remove .only

AlexBlokh committed:
fix attw

AlexBlokh committed:
o_o

AlexBlokh committed:
fix .existing filtering for views in crdb

AlexBlokh committed:
fix crdb drizzle talbes filtering

Sukairo-02 committed:
Returned old RLS syntax in deprecated form

Sukairo-02 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
fix postgres introspect with .existing / entity filters

AlexBlokh committed:
Merge commit 'e5cd57623f7caa3cb5f9c790cf6b24935b46b042' into alternation-engine

Sukairo-02 committed:
`column.as(alias)` prototype

Sukairo-02 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AndriiSherman committed:
Add index case

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

Sukairo-02 committed:
Test suite name fix, added tests for `column.as(alias)`

AlexBlokh committed:
+

AlexBlokh committed:
remove redundant default as object in postgres ddl

AlexBlokh committed:
Merge commit '485d8360f0d38246383ab7b9df6d48590c8562e2' into alternation-engine

Sukairo-02 committed:
Added `cockroach` casing tests

AndriiSherman committed:
add detect

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

Sukairo-02 committed:
View select fix for views with `column.as` in selection, integration tests for `column.as`

Sukairo-02 committed:
SQLite test case fix

Sukairo-02 committed:
Added missed drop clauses

Sukairo-02 committed:
Fixed wrong data type

Sukairo-02 committed:
Added missed table drops

Sukairo-02 committed:
Removed faulty `.catch`es

Sukairo-02 committed:
Added tables clearing

OleksiiKH0240 committed:
[integration-tests] updated singlestore tests

OleksiiKH0240 committed:
[integration-tests] decreased number of singlestore dbs

AlexBlokh committed:
update cockroach entity filters

- explain alternations partially implemented internally for pg and crdb
- fixed uniques in crdb, they're not always compatible with indexes,
  we need to only supply create table statement with them if they don't
  have a `where` and if their `method` is default
- crdb and pg column defaults are now string? in ddl

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

AlexBlokh committed:
fix crdb convertor

RomanNabukhotnyi committed:
Merge branch 'alternation-engine' into ae-api-studio

RomanNabukhotnyi committed:
refactor

RomanNabukhotnyi committed:
fix: Add api entries

RomanNabukhotnyi committed:
Merge pull request #5012 from drizzle-team/ae-api-studio

Prepare merge 'api-studio' to 'alternation-engine'

RomanNabukhotnyi committed:
Merge branch 'beta' into alternation-engine

RomanNabukhotnyi committed:
refactor workflows

L-Mario564 committed:
Update no-instanceof lint rule

AleksandrSherman committed:
[feat:psql]: explain command option

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AlexBlokh committed:
add sql highligher for cli and remove redundant strict for pg push

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

AleksandrSherman committed:
[feat]: mysql exlpain option

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AleksandrSherman committed:
[update]: prev merge conflict fix

Sukairo-02 committed:
Merge branch 'beta' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

Sukairo-02 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AleksandrSherman committed:
[update]: mssql ignore master system tables

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AleksandrSherman committed:
[fix]: more universal fix on prev commit

OleksiiKH0240 committed:
[integration-tests] added instrumentation.ts to sqlite; updated sqlite tests;

OleksiiKH0240 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

OleksiiKH0240 committed:
updated pnpm-lock.yaml

AlexBlokh committed:
properly log sql query errors, remove redundant strict flag

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

Sukairo-02 committed:
Redeclared broken imported enum

AlexBlokh committed:
Merge commit 'cd4c14cd1801162974f25e7f92c318e30e4bc919' into alternation-engine

AlexBlokh committed:
+

AlexBlokh committed:
+

AlexBlokh committed:
fix pg tests

Sukairo-02 committed:
Planetscale test cases fix

AlexBlokh committed:
+

AlexBlokh committed:
Merge commit '26b9925c9992b47a924b40c688566a6c76fd243d' into alternation-engine

AlexBlokh committed:
--init migration tests to end

Sukairo-02 committed:
Fixed PGJS migrator --init tests

Sukairo-02 committed:
Fixed pg-proxy migrate --init tests

AlexBlokh committed:
+

AlexBlokh committed:
Merge remote-tracking branch 'origin/alternation-engine' into alternation-engine

AlexBlokh committed:
+

AlexBlokh committed:
+

AlexBlokh committed:
+

AleksandrSherman committed:
[mssql]: introspect fix

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AleksandrSherman committed:
[update-kit]: explain for all dialects except sqlite

Sukairo-02 committed:
Removed typings from `cockroach-core`

Sukairo-02 committed:
Removed deprecated driver constructor overload

Sukairo-02 committed:
Fixed invalid typings removal logic

AleksandrSherman committed:
[sqlite]: explain

AleksandrSherman committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

AleksandrSherman committed:
[sqlite]: explain

AleksandrSherman committed:
[sqlite]: explain

AleksandrSherman committed:
[sqlite]: push - explain

AlexBlokh committed:
fix libsql for push command

Sukairo-02 committed:
Merge branch 'beta' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

Sukairo-02 committed:
Additional merge-related changes

RomanNabukhotnyi committed:
chore: Update SQLite connection message to include '@tursodatabase/database' as an option

RomanNabukhotnyi committed:
chore: Remove '@hono/bun-compress' dependency (Bun already supports CompressionStream)

Sukairo-02 committed:
Fixed missing bun drivers from kit, fixed lack of subquery procession in cockroach, mssql sessions, fixed broken test cases

Sukairo-02 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

RomanNabukhotnyi committed:
chore: Update SQLite connection message to include 'bun' as an option

RomanNabukhotnyi committed:
chore: Update driversPackages to include additional SQLite drivers

Sukairo-02 committed:
Fixed driver packages list

Sukairo-02 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

RomanNabukhotnyi committed:
fix: Add external dependencies for api build

Sukairo-02 committed:
Fixed broken test cases

Sukairo-02 committed:
Merge branch 'alternation-engine' of https://github.com/drizzle-team/drizzle-orm into alternation-engine

RomanNabukhotnyi committed:
fix: Update release routing logic

RomanNabukhotnyi committed:
fix: Correct variable usage in release routing logic

AlexBlokh committed:
+

AleksandrSherman committed:
[mysql]: suggestions

AndriiSherman committed:
Commutativity

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

Sukairo-02 committed:
Additional test cases

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

AndriiSherman committed:
Add pull relations to pg

AndriiSherman committed:
Merge branch 'alternation-engine' of github.com:drizzle-team/drizzle-orm into alternation-engine

AndriiSherman committed:
ignore commutative

AndriiSherman committed:
ignore commutative

AlexBlokh committed:
fix all tests issues

AlexBlokh committed:
remove .only

AlexBlokh committed:
postpone commutative tests

Eigilak commented:
When can we expect this big boi to be on beta? 🫶

AndriiSherman committed:
Up versions

AleksandrSherman committed:
[update]: codeql suggestion updates + new tests on generated

ThomOrlo commented on drizzle-orm/package.json:
```diff
@@ -70,6 +73,7 @@
 		"bun-types": "*",
 		"expo-sqlite": ">=14.0.0",
 		"gel": ">=2",
+		"mssql": "^11.0.1",
```
Given that `mssql` is only required for _Microsoft SQL Server_ functionalities, would it be better to declare it as `optional` in `peerDependenciesMeta`?

lkwr replied:
Created https://github.com/drizzle-team/drizzle-orm/pull/5089 for that
